pgml-extension/pgml_rust/src/backends/backend.rs

```diff
@@ -0,0 +1,42 @@
+use pgx::*;
+use serde::Deserialize;
+
+#[derive(PostgresEnum, Copy, Clone, Eq, PartialEq, Debug, Deserialize)]
+#[allow(non_camel_case_types)]
+pub enum Backend {
+    xgboost,
```

Comment:
Some of these backends don't differentiate between python/rust implementations, and seem to be confounded with the algorithm.

Reply:
You can pickle an xgboost class (and we do in 1.0). Native xgboost is serialized differently.